app/controllers/events_controller.rb

```diff
@@ -7,7 +7,11 @@ def index
 
   def show
     @event = Event.find(params[:id])
-    @registration = Registration.new(user: current_user, event: @event)
+    if current_user
+      @registration = Registration.first_or_initialize(user: current_user, event: @event)
+    else
+      @registration = Registration.new(user: current_user, event: @event)
```

Comment:
Isn't `current_user` nil? Is that still cool?

Reply:
Took `user: current_user` out of else case...that's dumb